CMakeLists.txt

```diff
@@ -1,24 +1,19 @@
-cmake_minimum_required(VERSION 3.8)
-project(bls12-381 VERSION 1.0.0)
+cmake_minimum_required(VERSION 3.12)
 
-#set(CMAKE_CXX_STANDARD 17)
-#set(CMAKE_CXX_EXTENSIONS ON)
-#set(CMAKE_CXX_STANDARD_REQUIRED ON)
-#add_compile_options(-Wall)
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR AND NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
 
-# compile with "-march" or enable cpu features explicitly for best performance:
-#set(CMAKE_CXX_FLAGS "-madx -mbmi2")
-set(CMAKE_CXX_FLAGS "-fstack-protector-strong")
```

Comment:
Should we leave the check `-fstack-protector-strong` enabled for debug builds?

Reply:
imo libraries meant to be included in a project like this should not be setting flags of this kind on their own. It should be set at the top level for consistency across the application.

In this particular case, I'm not even sure if it was meant to be left in. Originally when we were having trouble with the library failing in certain environments one configuration was a little odd because it worked for Matthias but not me. It turns out this was because [Arch ships their clang patched to always enable `-fstack-protector-strong`](https://gitlab.archlinux.org/archlinux/packaging/packages/clang/-/blob/main/enable-fstack-protector-strong-by-default.patch), presumably to enforce distro-wide hardening goals. So it's possible this could have been left over from just troubleshooting that problem.

Since we run with tests with ASAN, which should be a massive super set of what `-fstack-protector-strong` provides, I don't really think we need to expressly run a configuration with it. (and, of course, we are actually implicitly running it in CI since the Arch gcc & clang builds will have this enabled).